sound/soc/sof/intel/hda.c

```diff
@@ -1039,13 +1039,50 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 #endif
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
+/* Check if all slaves defined on the link can be found */
+static bool is_slave_found(struct snd_sof_dev *sdev,
+			   struct snd_soc_acpi_link *link,
+			   struct sdw_intel_ctx *sdw)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct sdw_intel_slave_id *ids = sdw->ids;
+	int num_slaves = sdw->num_slaves;
+	int part_id, link_id;
+	int j, k;
+
+	for (k = 0; k < link->dev_num; k++) {
+		part_id = (link->adr[k] >> 8) & GENMASK(15, 0);
+		for (j = 0; j < num_slaves; j++) {
+			if (ids[j].id.part_id == part_id) {
+				link_id = sdw->ids[j].link_id;
+				if (link->mask & BIT(link_id)) {
+					dev_dbg(bus->dev,
```

Comment:
if you use many if, you may get into trouble of exceeding 80 chars.
you can do it as:
````
if (ids[j].id.part_id != part_id)
      continue;
````